resolver/resolver.cc

```diff
@@ -1259,7 +1259,7 @@ class ResolveTypeParamsWalk {
         Timer timeit(ctx.state.tracer(), "resolver.type_params");
 
         for (auto &tree : trees) {
-            tree.tree = ast::TreeMap::apply(ctx, params, std::move(tree.tree));
+            tree.tree = ast::ShallowMap::apply(ctx, params, std::move(tree.tree));
```

Comment:
I'm not entirely comfortable with not having ENFORCE's that make sure that this found all the usages.
Can we add some postconditions?